src/interceptors/fetch/index.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  flattenHeadersObject,
+  Headers,
+  headersToObject,
+  objectToHeaders,
+} from 'headers-utils'
+import {
+  Interceptor,
+  IsomoprhicRequest,
+  IsomoprhicResponse,
+  MockedResponse,
+} from '../../createInterceptor'
+
+const debug = require('debug')('fetch')
+
+export const interceptFetch: Interceptor = (observer, resolver) => {
+  const pureFetch = window.fetch
+
+  debug('replacing "window.fetch"...')
+
+  window.fetch = async (input, init) => {
+    const ref = new Request(input, init)
+    const url = typeof input === 'string' ? input : input.url
+    const method = init?.method || 'GET'
+
+    debug('[%s] %s', method, url)
+
+    const request: IsomoprhicRequest = {
+      url: new URL(url, location.origin),
+      method: method,
+      headers: init?.headers ? headersToObject(new Headers(init.headers)) : {},
+      body: await ref.text(),
```

Comment:
This part is extremely frustrating to test as non-string values must be serialized when passing from Playwrigth (context where the request has been intercepted) and the test runner (Jest, in Node.js). 

@marcosvega91, do you think this kind of body conversion will work for request bodies provided as `Blob`/`ArrayBuffer`/`FormData`/`ReadableStream`/etc.?

Reply:
`ReadableStream` is available only in the response. I think that in this case there will be no problem